### PR TITLE
v7.21.0

### PR DESCRIPTION
**Pull request by cherniavskii**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,92 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.20.1
+
+_Oct 17, 2024_
+
+We'd like to offer a big thanks to the 13 contributors who made this release possible. Here are some highlights ✨:
+
+- 💫 `dataset` prop support for `ScatterChart` component
```
Comment (noraleonte) [inline CHANGELOG.md:12]:
```suggestion
- 💫 Added `dataset` prop support for `ScatterChart` component
```

**Review by noraleonte — APPROVED** [review]
LGTM 👀 👌

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,92 @@
...
+- 🐞 Bugfixes
+- 📚 Documentation improvements
+
+Special thanks go out to the community contributors who have helped make this release possible:
+@arminmeh, @k-rajat19, @kalyan90, @romgrk, @rotembarsela, @wangkailang.
```
Comment (alexfauquette) [inline CHANGELOG.md:17]: I think @arminmeh and @romgrk are team member 😇😁

**Review by alexfauquette — APPROVED** [review]

**Comment by flaviendelangle** [comment]
I'd like to have #15002 released this week :pray:

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,92 @@
...
+We'd like to offer a big thanks to the 13 contributors who made this release possible. Here are some highlights ✨:
+
+- 💫 Added `dataset` prop support for `ScatterChart` component
+- 🐞 Bugfixes
+- 📚 Documentation improvements
+
+Special thanks go out to the community contributors who have helped make this release possible:
+@k-rajat19, @kalyan90, @rotembarsela, @wangkailang.
+Following are all team members who have contributed to this release:
+@arthurbalduini, @cherniavskii, @flaviendelangle, @JCQuintas, @LukasTy, @MBilalShafi, @arminmeh, romgrk.
```
Comment (arminmeh) [inline CHANGELOG.md:19]:
```suggestion
@arthurbalduini, @cherniavskii, @flaviendelangle, @JCQuintas, @LukasTy, @MBilalShafi, @arminmeh, @romgrk.
```

**Review by arminmeh — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,92 @@
...
+- [docs] Add recipe showing how to toggle detail panels on row click (#14666) @k-rajat19
+- [docs] Fix broken link to the validation section in the Data grid component (#14973) @arminmeh
+- [docs] Update v5 migration codesandbox @oliviertassinari
+
+### Core
+
+- [core] Fix shortcut with localization keyboard (#14220) @rotembarsela
+- [core] Fix docs deploy command (#14920) @arminmeh
+- [code-infra] Prepare some tests to work in `vitest/playwright` (#14926) @JCQuintas
+- [Data grid] Fix excessive white space at the end of grid (#14864) @kalyan90
```
Comment (flaviendelangle) [inline CHANGELOG.md:88]: Need to be moved inside the right section

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,92 @@
...
+
+Same changes as in `@mui/x-data-grid-pro@7.20.1`.
+
+### Date and Time Pickers
+
+#### `@mui/x-date-pickers@7.20.1`
+
+- [pickers] Cleanup `PageUp` and `PageDown` event handlers on time components (#14928) @arthurbalduini
+- [pickers] Create the new picker's `ownerState` object (#14889) @flaviendelangle
+- [pickers] Fix `PickerValidDate` usage in `DateRangePickerToolbar` (#14925) @flaviendelangle
```
Comment (flaviendelangle) [inline CHANGELOG.md:49]:
```suggestion
- [pickers] Fix `PickerValidDate` usage in `<DateRangePickerToolbar />` (#14925) @flaviendelangle
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,92 @@
...
+
+#### `@mui/x-date-pickers-pro@7.20.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-date-pickers@7.20.1`.
+
+### Charts
+
+#### `@mui/x-charts@7.20.1`
+
+- [charts] Allow `dataset` to be used with `ScatterChart` (#14915) @JCQuintas
```
Comment (flaviendelangle) [inline CHANGELOG.md:59]:
```suggestion
- [charts] Allow `dataset` to be used with `<ScatterChart />` (#14915) @JCQuintas
```

File: CHANGELOG.md (hunk 4 above)
Comment (flaviendelangle) [reply CHANGELOG.md:88]: And to follow the style guide:

```suggestion
- [Data grid] Fix excessive white space at the end of the Data Grid (#14864) @kalyan90
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,92 @@
...
+
+#### `@mui/x-charts-pro@7.0.0-beta.5` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-charts@7.20.1`.
+
+### Tree View
+
+#### `@mui/x-tree-view@7.20.1`
+
+- [TreeView] Fix `alpha` usage with CSS variables (#14969) @wangkailang
```
Comment (flaviendelangle) [inline CHANGELOG.md:70]:
```suggestion
- [TreeView] Fix `alpha` usage with CSS variables (#14969) @wangkailang
- [TreeView] Fix hydration error (#15002) @flaviendelangle 
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -35,6 +35,7 @@ Same changes as in `@mui/x-data-grid@7.20.1`, plus:
 
 - [DataGridPro] Fix indeterminate checkbox state for server-side data (#14956) @MBilalShafi
 - [DataGridPro] Fix scrolling performance when `rowHeight={undefined}` (#14983) @cherniavskii
+- [DataGridPro] List View (#14393) @KenanYusuf @cherniavskii
```
Comment (flaviendelangle) [inline CHANGELOG.md:39]: That's a short PR title :laughing:

**Review by flaviendelangle — APPROVED** [review]

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,94 @@
...
+- [TreeView] Fix usage of the `aria-selected` attribute (#14991) @flaviendelangle
+- [TreeView] Fix hydration error (#15002) @flaviendelangle
+
+### `@mui/x-codemod@7.20.1`
+
+- [codemod] Add a new utility to rename imports (#14919) @flaviendelangle
+
+### Docs
+
+- [docs] Add recipe showing how to toggle detail panels on row click (#14666) @k-rajat19
```
Comment (flaviendelangle) [inline CHANGELOG.md:82]:
```suggestion
- [docs] Add recipe showing how to toggle detail panels on row click (#14666) @k-rajat19
- [docs] Enforce component style rules for the Tree View (#14963) @samuelsycamore 
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,94 @@
...
+We'd like to offer a big thanks to the 13 contributors who made this release possible. Here are some highlights ✨:
+
+- 💫 Added [`dataset` prop support for the Scatter Chart component](https://mui.com/x/react-charts/scatter/#using-a-dataset)
+- 🐞 Bugfixes
+- 📚 Documentation improvements
+
+Special thanks go out to the community contributors who have helped make this release possible:
+@k-rajat19, @kalyan90, @rotembarsela, @wangkailang.
+Following are all team members who have contributed to this release:
+@arthurbalduini, @cherniavskii, @flaviendelangle, @JCQuintas, @LukasTy, @MBilalShafi, @arminmeh, @romgrk, @KenanYusuf, @oliviertassinari.
```
Comment (flaviendelangle) [inline CHANGELOG.md:19]:
```suggestion
@arthurbalduini, @cherniavskii, @flaviendelangle, @JCQuintas, @LukasTy, @MBilalShafi, @arminmeh, @romgrk, @KenanYusuf, @oliviertassinari, @samuelsycamore.
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -3,6 +3,94 @@
...
+- [DataGrid] Fix scroll jumping (#14929) @romgrk
+- [DataGrid] Fix excessive white space at the end of the Data Grid (#14864) @kalyan90
+
+#### `@mui/x-data-grid-pro@7.20.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-data-grid@7.20.1`, plus:
+
+- [DataGridPro] Fix indeterminate checkbox state for server-side data (#14956) @MBilalShafi
+- [DataGridPro] Fix scrolling performance when `rowHeight={undefined}` (#14983) @cherniavskii
+- [DataGridPro] List View (#14393) @KenanYusuf @cherniavskii
```
Comment (KenanYusuf) [inline CHANGELOG.md:39]:
```suggestion
- [DataGridPro] List view (#14393) @KenanYusuf @cherniavskii
```

**Review by KenanYusuf — APPROVED** [review]

File: CHANGELOG.md (hunk 8 above)
Comment (arminmeh) [reply CHANGELOG.md:39]: does this require minor bump for the release?

Comment (cherniavskii) [reply CHANGELOG.md:39]: We don't announce it publicly yet, we still need docs

Comment (cherniavskii) [reply CHANGELOG.md:39]: But a minor bump is still a good idea, I think I'll do that

**Pull request merged**